### PR TITLE
Distributed train utils

### PR DESCRIPTION
**Pull request by kingrc15**
- Add GatherLayer for gathering logits from multiple devices with backprop
 - EBCL now uses DDP and sync batch norms
 - Fixed on_validation_epoch_end

### CODE REVIEW

**Comment by kingrc15** [comment]
testing when doing ddp training has been removed. https://github.com/Lightning-AI/pytorch-lightning/issues/12862

**Review by Oufattole — APPROVED** [review]
LGTM
